react/w-books/src/app/routers/AppRouter.tsx

```diff
@@ -1,18 +1,23 @@
 import React from 'react';
-import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import { BrowserRouter as Router, Switch } from 'react-router-dom';
 
-import PATHS from '~constants/paths';
-import Home from '~screens/Home';
-import Login from '~screens/Login';
-import SignUp from '~screens/SignUp';
+import Home from '../screens/Home';
+import Login from '../screens/Login';
+import SignUp from '../screens/SignUp';
+import BookDetail from '../components/BookDetail/index';
 
+import PrivateRoute from './PrivateRoute';
+import PublicRoute from './PublicRoute';
+
+const isAuthenticated = localStorage.getItem('access-token');
 function AppRouter() {
   return (
     <Router>
       <Switch>
-        <Route path={PATHS.signUp} component={SignUp} exact />
-        <Route path={PATHS.root} component={Login} exact />
-        <Route path={PATHS.home} component={Home} exact />
+        <PublicRoute component={SignUp} path="/sign_up" isSignedIn={!!isAuthenticated} />
+        <PrivateRoute component={Home} path="/home" isSignedIn={!!isAuthenticated} />
+        <PrivateRoute component={BookDetail} path="/books/:id" isSignedIn={!!isAuthenticated} />
+        <PublicRoute component={Login} exact path="/" isSignedIn={!!isAuthenticated} />
```

Comment:
The PrivateRoute component and the PublicRoute component are equals, you can make an only customRoute component, and here do the validations